hyperhtml.js

```diff
@@ -278,15 +305,61 @@ var hyperHTML = (function () {'use strict';
   // Helpers
   // -------------------------
 
-  // it does exactly what it says
-  function appendNodes(node, childNodes) {
-    for (var
-      i = 0,
-      length = childNodes.length;
-      i < length; i++
-    ) {
-      node.appendChild(childNodes[i]);
+  // given a generic text or comment node
+  // remove the surrounding emptiness
+  function cleanAround(textNode, sibling) {
+    var adj = textNode[sibling];
+    if (adj && adj.nodeType === 3 && trim.call(adj.textContent).length < 1) {
```

Comment:
I need a global Node first to do that ... remember this project is used also in [nativeHTML](https://github.com/WebReflection/nativeHTML) through [basicHTML](https://github.com/WebReflection/basicHTML) and the less I assume globals the better it is.

It's already tricky to initialize `hyperHTML` in `basicHTML` due one off feature detection using `document`: https://github.com/WebReflection/basicHTML/blob/master/test/hyperhtml.js